From 063cd86c18b9815a76549c4ad5e82103898ade23 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Edwin=20T=C3=B6r=C3=B6k?= <edwin@etorok.net>
Date: Tue, 8 Jun 2021 20:32:23 +0000
Subject: [PATCH] asmcomp/dune: fix build on RISC-V
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

dune build @libs was failing with
```
        bash asmcomp/CSE.ml,asmcomp/arch.ml,asmcomp/proc.ml,asmcomp/reload.ml,asmcomp/scheduling.ml,asmcomp/selection.ml (exit 1)
(cd _build/default/asmcomp && /bin/bash -e -u -o pipefail -c 'cp `grep '\''^ARCH='\'' ../Makefile.config | cut -d'\''='\'' -f2`/*.ml .')
cp: cannot stat 'riscv/*.ml': No such file or directory
        bash asmcomp/contains-input-name,asmcomp/emit.ml (exit 1)
(cd _build/default/asmcomp && /bin/bash -e -u -o pipefail -c '../tools/cvt_emit.exe < `cat contains-input-name`') > _build/default/asmcomp/emit.ml
/bin/bash: line 1: riscv/emit.mlp: No such file or directory
```

No change entry needed

Signed-off-by: Edwin Török <edwin@etorok.net>
---
 asmcomp/dune | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/asmcomp/dune b/asmcomp/dune
index a208b56e5..1a4d561d6 100644
--- a/asmcomp/dune
+++ b/asmcomp/dune
@@ -21,6 +21,7 @@
           (glob_files arm64/*.ml)
           (glob_files i386/*.ml)
           (glob_files power/*.ml)
+          (glob_files riscv/*.ml)
           (glob_files s390x/*.ml))
  (action  (bash "cp `grep '^ARCH=' %{conf} | cut -d'=' -f2`/*.ml .")))
 
@@ -33,6 +34,7 @@
           arm64/emit.mlp
           i386/emit.mlp
           power/emit.mlp
+          riscv/emit.mlp
           s390x/emit.mlp)
  (action
    (progn
-- 
2.39.0.windows.2

